src/main/java/com/mkopec/ticketbookingapp/repository/ReservationRepository.java

```diff
@@ -0,0 +1,10 @@
+package com.mkopec.ticketbookingapp.repository;
+
+import com.mkopec.ticketbookingapp.domain.Reservation;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ReservationRepository extends JpaRepository<Reservation, Long> {
```

Comment:
jaki jest problem z takim repository?

Reply:
Sądzę, że chodzi to o to, że JpaRepository jest przeznaczone do relacyjnych baz danych

Reply:
nie, chodzi bardziej o pewne zasady, dobre praktyki projektowania/programowania

Reply:
W repozytorium Jpa jest dużo metod związanych z zapisywaniem i usuwaniem a nie zawsze jest to potrzebne po stronie serwisu, a nawet niebezpieczne. Interfejs powoduje, że te metody są one publiczne. 
Powinniśmy zapewnić, aby nie były dostępne do stronie serwisu, aby ktoś nie wywołał metody np. deleteAll() albo save nadpisując jakieś dane.
W takim repozytorium można definiować zapytanie na podstawie nazwy metody, np. findAllByName(), ale zmiana nazwy atrybutu name powoduje już niedziałanie takiej metody. Zmiana repozytorium może być wtedy problematyczna z powodu tych metod repozytorium